File: apps/java/io/dustin/apps/board/domain/community/posting/repository/custom/CustomPostingRepository.java
package io.dustin.apps.board.domain.community.posting.repository.custom;

import io.dustin.apps.board.domain.community.posting.model.dto.PostingDto;
import io.dustin.apps.board.domain.community.posting.model.dto.PostingListDto;

import java.util.List;

public interface CustomPostingRepository {

    PostingDto getPosting(long userId, long postingId);

    List<PostingListDto> getPostingList(long userId, Long nextId, int size);

    PostingDto getMyPosting(long userId, long postingId);

    List<PostingListDto> getMyPostingList(long userId, Long nextId, int size);

}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/community/posting/repository/custom/CustomPostingRepository.kt
package io.dustin.apps.board.domain.community.posting.repository.custom

import io.dustin.apps.board.domain.community.posting.model.dto.PostingDto
import io.dustin.apps.board.domain.community.posting.model.dto.PostingListDto

interface CustomPostingRepository {
    fun getPosting(userId: Long, postingId: Long): PostingDto
    fun getPostingList(userId: Long, recordsCount: Long, nextId: Long?,): List<PostingListDto>
    fun getMyPosting(userId: Long, postingId: Long): PostingDto
    fun getMyPostingList(userId: Long, nextId: Long?, size: Long): List<PostingListDto>
}
